docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,130 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom {ref}/analysis.html[Lucene 
+analyzers, token filters, character filters, and tokenizers]. 
+
+[discrete]
+==== Plugin file structure
+
+Plugins are ZIP files composed of JAR files and two metadata files:
+
+* `stable-plugin-descriptor.properties` - a Java properties file that describes 
+the plugin. Refer to <<plugin-descriptor-file-{plugin-type}>>.
+* `named_components.json` - a JSON file mapping interfaces to key-value pairs 
+of component names and implementation classes.
+
+Note that only JAR files at the root of the plugin are added to the classpath
+for the plugin. If you need other resources, package them into a resources JAR.
+
+[discrete]
+==== The analysis plugin API
+
+Text analysis plugins can be written against the analysis plugin API. This API
+consists of the following dependencies:
+
+* `plugin-api`- an API used by plugin developers to implement custom {es} 
+plugins (stable).
+* `plugin-analysis-api` - an API used by plugin developers to implement analysis
+plugins and integrate them into {es}.
+* `lucene-analysis-common` - a dependency of `plugin-analysis-api` that contains
+core Lucene analysis interfaces like `Tokenizer`, `Analyzer`, and `TokenStream`.
+
+The advantage of using this API is that the resulting plugin is
+forward-compatible with new minor versions of {es}. While elements may be added
```

Comment:
I don't think forward compatibility is the terminology we should use. Rather, Elasticsearch is maintaining compatibility of the APIs, such that newer versions of the API (in newer versions of Elasticsearch) are backwards compatible.